From e407b9928506bd85e72d072a9e3d0d8a77c15257 Mon Sep 17 00:00:00 2001
From: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Date: Fri, 3 Feb 2017 19:32:44 +0200
Subject: [PATCH 1505/5242] MLK-13839 fbdev: mxsfb: Fix null pointer
 dereference

commit  86f7e3bedac0c3e3207d4699fc0032a562801d22 from
https://source.codeaurora.org/external/imx/linux-imx.git

In function mxsfb_check_var an explicit null pointer dereference occurs
when input frame is 32 bpp (var->bits_per_pixel) and the output frame is
different from 8/16/18/24 (host->ld_intf_width).

Even though a 32bpp output is possible under certain conditions, this is
currently not implemented, so any other output bpp value is not valid.

Signed-off-by: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 7fa2a09..4022cae 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -4,6 +4,7 @@
  * This code is based on:
  * Author: Vitaly Wool <vital@embeddedalley.com>
  *
+ * Copyright 2017 NXP
  * Copyright 2008-2015 Freescale Semiconductor, Inc. All Rights Reserved.
  * Copyright 2008 Embedded Alley Solutions, Inc All Rights Reserved.
  *
@@ -632,6 +633,15 @@ static int mxsfb_check_var(struct fb_var_screeninfo *var,
 			/* real 24 bit */
 			rgb = def_rgb888;
 			break;
+		default:
+			/*
+			 * 32-bit output is possible through I/O muxing, if this
+			 * option is available on chip. Currently not
+			 * implemented.
+			 */
+			pr_debug("Currently unsupported output colour depth: %u\n",
+				 host->ld_intf_width);
+			return -EINVAL;
 		}
 		break;
 	default:
-- 
1.7.9.5

